src/main/java/org/magicdgs/http/jsr203/HttpAbstractFileSystemProvider.java

```diff
@@ -35,27 +46,72 @@
     @Override
     public abstract String getScheme();
 
+    // check the conditions for an URI, and return it if it is correct
+    private URI checkUri(final URI uri) {
+        // non-null URI
+        Utils.nonNull(uri, () -> "null URI");
+        // non-null authority
+        Utils.nonNull(uri.getAuthority(),
+                () -> String.format("%s requires URI with authority: invalid %s", this, uri));
+        // check the scheme (sanity check)
+        if (!getScheme().equalsIgnoreCase(uri.getScheme())) {
+            throw new ProviderMismatchException(String.format("Invalid scheme for %s: %s",
+                    this, uri.getScheme()));
+        }
+        return uri;
+    }
+
     @Override
     public final HttpFileSystem newFileSystem(final URI uri, final Map<String, ?> env)
             throws IOException {
-        throw new UnsupportedOperationException("Not implemented");
+        checkUri(uri);
+
+        if (fileSystems.containsKey(uri.getAuthority())) {
+            throw new FileSystemAlreadyExistsException("URI: " + uri);
+        }
+
+        return fileSystems.computeIfAbsent(uri.getAuthority(),
+                (auth) -> new HttpFileSystem(this, auth));
     }
 
     @Override
     public final HttpFileSystem getFileSystem(final URI uri) {
-        throw new UnsupportedOperationException("Not implemented");
+        final HttpFileSystem fs = fileSystems.get(checkUri(uri).getAuthority());
+        if (fs == null) {
+            throw new FileSystemNotFoundException("URI: " + uri);
+        }
+        return fs;
     }
 
     @Override
-    public final Path getPath(final URI uri) {
-        throw new UnsupportedOperationException("Not implemented");
+    public final HttpPath getPath(final URI uri) {
+        checkUri(uri);
+        return fileSystems
+                .computeIfAbsent(uri.getAuthority(), (auth) -> new HttpFileSystem(this, auth))
+                .getPath(uri);
     }
 
     @Override
     public final SeekableByteChannel newByteChannel(final Path path,
             final Set<? extends OpenOption> options, final FileAttribute<?>... attrs)
             throws IOException {
-        throw new UnsupportedOperationException("Not implemented");
+        Utils.nonNull(path, () -> "null path");
+        Utils.nonNull(options, () -> "null options");
+
```

Comment:
Add comment saying that the URL is only checked if the options are correct, below with `checkUri`

Reply:
Done